### ISSUE TITLE
[BUG] Stripe-Min cocoapod fails on Xcode 14

### ISSUE THREAD

**Issue by dingwilson** [stripe/stripe-ios#2013]
## Summary

stripe-min is currently throwing the same error as shown in #1973 

```
in STPApplePayContext.swift l:240

'(any PKPaymentAuthorizationControllerDelegate) -> ((PKPaymentAuthorizationController, PKShippingMethod, @escaping (PKPaymentRequestShippingMethodUpdate) -> Void) -> Void)?' is not convertible to 'pkDidSelectShippingMethodSignature' (aka 'Optional<((PKPaymentAuthorizationControllerDelegate) -> (PKPaymentAuthorizationController, PKShippingMethod, @escaping (PKPaymentRequestShippingMethodUpdate) -> ()) -> ())>')
```

Is it possible to cut an updated compact-sdk version that supports Xcode 14?

## Code to reproduce

n/a

## iOS version

All versions

## Installation method

Cocoapods

## SDK version

pod 'Stripe-Min', :git => 'https://github.com/stripe/stripe-ios.git', :branch => 'compact-sdk-21.6.0'

## Other information

n/a

**Comment by davidme-stripe** [comment]
Hi @dingwilson, thanks for the request!

The latest SDK contains [StripeApplePay](https://github.com/stripe/stripe-ios/tree/master/StripeApplePay), a separate minimal module just for Apple Pay. To adopt it, follow the instructions in [our Apple Pay guide](https://stripe.com/docs/apple-pay?platform=ios#migrating-from-stpapplepaycontext).

If you're not ready to migrate away from the `Stripe-Min` prototype, I'd recommend forking and applying this patch to the `Stripe-Min` pod: https://github.com/stripe/stripe-ios/commit/c787ef622ce7b540040b225017532a5cb8b77963

Let us know if there's anything else we can do to help.

**Comment by jerometonnelier** [comment]
Hi @davidme-stripe,
Quick question : I'm having the same issue but with SPM. Is there any way to get a fix soon or is there a branch I can use ? 
Thanks

**Comment by dingwilson** [comment]
Thanks for the fast reply @davidme-stripe!

+1 to @jerometonnelier 's request - Is it possible for us to get a stripe-owned branch/version of the `Stripe-Min` prototype + patch? We will look to migrate to the latest StripeApplePay SDK soon, but right now it would be really helpful if we can use a minimal-change version of the existing SDK that can unblock us from migrating our dev team to Xcode 14.

**Comment by davidme-stripe** [comment]
We've pushed this to [`compact-sdk-21.6.0-xcode14`](https://github.com/stripe/stripe-ios/compare/compact-sdk-21.6.0-xcode14). Let me know if that works for you!

@jerometonnelier, the `Stripe-Min` pod isn't supported on SPM. Which version of the SDK are you using?

**Comment by dingwilson** [comment]
Looks like that worked! Thank you so much @davidme-stripe!

**Comment by jerometonnelier** [comment]
@davidme-stripe Thx. I'lm not using the min SDK? just the classic one [Stripe](https://github.com/stripe/stripe-ios/blob/master/Package.swift) ;-)

**Comment by davidme-stripe** [comment]
@jerometonnelier, I think the issue you mentioned was resolved in 22.6.0. I'd recommend updating to the latest version, [22.8.1](https://github.com/stripe/stripe-ios/releases/tag/22.8.1).

**Comment by jerometonnelier** [comment]
it is indeed @davidme-stripe, my bad 😓Thx for your help

**Comment by davidme-stripe** [comment]
Hi @dingwilson,

We released SDK 23.0.0 today, which adds separate `StripePayments`, `StripePaymentsUI`, and `StripeApplePay` modules. I'd recommend migrating your App Clip to use these if you can. Thanks!
